PowerEditor/src/MISC/Common/Common.h

```diff
@@ -205,3 +207,34 @@ std::string ws2s(const std::wstring& wstr);
 bool deleteFileOrFolder(const generic_string& f2delete);
 
 void getFilesInFolder(std::vector<generic_string>& files, const generic_string& extTypeFilter, const generic_string& inFolder);
+
+template<typename T> size_t VecRemoveDuplicates(std::vector<T>& vec, bool isSorted = false, bool canSort = false)
+{
+	if (!isSorted && canSort)
+	{
+		std::sort(vec.begin(), vec.end());
+		isSorted = true;
+	}
+
+	if (isSorted)
+	{
+		typename std::vector<T>::iterator it;
+		it = std::unique(vec.begin(), vec.end());
+		vec.resize(distance(vec.begin(), it));  // unique() does not shrink the vector
+	}
+	else
+	{
+		std::unordered_set<T> seen;
+		auto newEnd = std::remove_if(vec.begin(), vec.end(), [&seen](const T& value)
+			{
+				if (seen.find(value) != std::end(seen))
+				{
+					return true;
+				}
+				seen.insert(value);
+				return false;

```

Comment:
This could be simplified to ```return !seen.insert(value).second;``` since the second return value of insert is if the insertion happened. It would avoid the double-lookup cost of check-if-exists then insert-if-not.

Reply:
For a 1M-line file of 20 random words making up each line (character count: 119376024, word count: 20043630), with a random 15% of lines duplicated by appending, then the file randomly sorted, then finally duplicates removed with this new feature, I saw these timings:

* my original code in this PR: **6.852 s**
* the suggested change to the code: **5.936 s**

File compares of the output were identical, so I will adopt this suggested change as a performance-enhancer.
Thanks!